### PR TITLE
Tests for solvers + minos & dqq solver support

### PR DESCRIPTION
**Pull request by laurentheirendt**
**Tests**
- `MIQP`
- `MPS` (write and read)
- tests for LP solvers

**Solver support**
- `DQQ`
- `quadMINOS`
- `pdco`

*Note*: commit of `binary` submodule updated.

**I hereby confirm that I have:**

- [X] Selected `develop` as a target branch (top left drop-down menu)
- [X] Tested my code on my own machine
- [X] Followed the guidelines in the [Contributing Guide](https://github.com/opencobra/cobratoolbox/blob/master/.github/CONTRIBUTING.md)

### CODE REVIEW

File: src/solvers/solveCobraMILP.m [hunk 1]
```diff
@@ -81,118 +81,135 @@
...
+if ~isempty(varargin)
+    isdone = false(size(varargin));
+
+    if strcmp(varargin{1}, 'default')  % Set tolerances to COBRA toolbox defaults
+        [feasTol, optTol] = getCobraSolverParams('LP', optParamNames(5:6), 'default');
+        isdone(1) = true;
+        varargin = varargin(~isdone);
+
+    elseif isstruct(varargin{1})  % solver-specific parameter structure
+        if isstruct(varargin{1})
```
Comment (syarra) [inline src/solvers/solveCobraMILP.m]: Redundant `if`

File: src/solvers/solveCobraLP.m [hunk 2]
```diff
@@ -247,11 +264,18 @@
...
+algorithm = 'default';
 
 t_start = clock;
 switch solver
     case 'opti'
+        if isunix
+            error('OPTI is not compatible with UNIX systems (macOS or Linux).')
+        end
+
+        if verLessThan('matlab', '8.4')
```
Comment (syarra) [inline src/solvers/solveCobraLP.m]: Shouldn't it be `if ~verLessThan('matlab', '8.4')`?

File: src/solvers/solveCobraMILP.m (hunk 1 above)
Comment (laurentheirendt) [inline src/solvers/solveCobraMILP.m]: Fixed.

File: src/solvers/solveCobraLP.m (hunk 2 above)
Comment (laurentheirendt) [inline src/solvers/solveCobraLP.m]: Fixed.

File: src/solvers/changeCobraSolver.m [hunk 3]
```diff
@@ -275,9 +287,13 @@
...
         solverOK = checkSolverInstallationFile(solverName, 'BuildMPS', printLevel);
     case 'quadMinos'
-        solverOK = checkSolverInstallationExecutable(solverName, 'minos', true, printLevel);
+        if isunix
+            solverOK = checkSolverInstallationFile(solverName, 'minos', printLevel);
+        end
     case 'dqqMinos'
-        solverOK = checkSolverInstallationExecutable(solverName, 'run1DQQ', true, printLevel);
+        if isunix
+            solverOK = checkSolverInstallationExecutable(solverName, 'run1DQQ', true, printLevel);
```
Comment (laurentheirendt) [inline src/solvers/changeCobraSolver.m]: I removed `checkSolverInstallationExecutable` as not relevant anymore.

**Pull request merged**